README.md

```diff
@@ -35,3 +35,97 @@ steps should work:
 
 You should now have a working `build/Release/CheriVis.app`.  For more complex
 build configurations or to edit the code, open the XCode project.
+
+### Building on Linux (Ubuntu 16.04.4 LTS Xenial Xerus)
+#### Cheritrace
+On Linux, first you must build the cheritrace library, e.g. by doing
+
+    $ cd cheritrace
+    $ mkdir Build
+    $ cd Build
+    $ # Make sure you have a recent clang
+    $ cmake .. -GNinja -DCMAKE_CXX_COMPILER=/usr/bin/clang++ -DCMAKE_C_COMPILER=/usr/bin/clang -DLLVM_CONFIG=/path/to/cheri/sdk/bin/llvm-config
+    $ ninja
+
+If you get a message about
+`/path/to/cheri/sdk/lib/libLLVMDebugInfoDWARF.a` being the wrong
+format (and if you do `file
+/path/to/cheri/sdk/lib/libLLVMDebugInfoDWARF.a` and it says it is
+empty, you need a CHERI-SDK which has that file, e.g. by using
+`cheribuild.py` (from github).
+
+
+#### Modern GNUstep (libobjc2)
+The Ubuntu GNUstep stuff is too old, so these steps install a GNUstep
+system under `/usr/GNUstep/`.
+
+    mkdir GNUstep-build
+    cd GNUstep-build
+
+    export CC=clang
+    export CXX=clang++
+
+    mkdir -p libobjc2 && wget -qO- https://github.com/gnustep/libobjc2/archive/v1.8.1.tar.gz | tar xz -C libobjc2 --strip-components=1
+    mkdir -p make && wget -qO- ftp://ftp.gnustep.org/pub/gnustep/core/gnustep-make-2.6.8.tar.gz | tar xz -C make --strip-components=1
+    mkdir -p base && wget -qO- ftp://ftp.gnustep.org/pub/gnustep/core/gnustep-base-1.24.9.tar.gz | tar xz -C base --strip-components=1
+    mkdir -p gui && wget -qO- ftp://ftp.gnustep.org/pub/gnustep/core/gnustep-gui-0.25.0.tar.gz | tar xz -C gui --strip-components=1
+    mkdir -p back && wget -qO- ftp://ftp.gnustep.org/pub/gnustep/core/gnustep-back-0.25.0.tar.gz | tar xz -C back --strip-components=1
+
+    cd make
+    ./configure --enable-debug-by-default --with-layout=gnustep --enable-objc-nonfragile-abi --enable-objc-arc
+    make -j8
+    sudo -E make install
+    cd ..
+
+    . /usr/GNUstep/System/Library/Makefiles/GNUstep.sh
+
+    cd libobjc2
```

Comment:
These are the wrong way around.  You should build libobjc2 and then build GNUstep Make.

Reply:
This is the building the gnustep make twice, as without it, the libobjc2 files seemed to end up in /usr/local and the gnustep.org files in /usr/GNUstep/, there is probably a simpler way to do it, but building gnustep-make then sourcing the shell file seems to make things work.

Reply:
Don't do that.  As I said, it uses CMAKE_INSTALL_PREFIX as the prefix without GNUstep Make.  If you want that to be /usr, rather than /usr/local, then set it.

Reply:
Okay, that part seems to work with the /usr prefix (I think I misunderstood you to mean to use the --with-layout=fhs stuff)

Reply:
Actually, I just used the wrong shell, which still had variables from sourcing the GNUstep.sh script

Using -DCMAKE_INSTALL_PREFIX=/usr installs stuff under /usr/lib and /usr/include, which is what I have had problems when I was trying to do it on Friday